File: apps/api/aqapi/icon_category/domain/value_object/icon_category_id.py
from dataclasses import dataclass

from aqapi.core.domain.value_object.base_id import BaseId

@dataclass
class IconCategoryId(BaseId):
    """アイコンカテゴリのIDを表すクラス"""
    
    def __init__(self, value: int):
        super().__init__(value)

File: packages/backend/src/features/icon-category/value-objects/iconCategoryId.ts
import { BaseId, BaseIdSchema } from '@backend/core/value-object/base_id';
import { z } from 'zod';

/**
 * IconCategoryIdのZodスキーマ
 */
export const IconCategoryIdSchema = BaseIdSchema;

/**
 * アイコンカテゴリのIDを表すクラス
 */
export class IconCategoryId extends BaseId {
  constructor(value: number) {
    super(value);
  }

  /**
   * IDの名前を返す
   */
  protected get _valueName(): string {
    return `IconCategoryId(${this._value})`;
  }

  /**
   * Zodスキーマから新しいIconCategoryIdインスタンスを作成
   * @param data Zodスキーマに準拠したデータ
   */
  static fromZodData(data: z.infer<typeof BaseIdSchema>): IconCategoryId {
    return new IconCategoryId(data);
  }
}
